### DISCUSSION TITLE
OpenHistoricalMap

### DISCUSSION THREAD

**Discussion by 1ec5** [valhalla/valhalla#4593] [Ideas]
Hello from the [OpenHistoricalMap](https://wiki.openstreetmap.org/wiki/OpenHistoricalMap) project! Yes, you heard right, we’re going to route autonomous vehicles over long-gone roads that have been built over, and we’re going to do it based on user-generated content. Just kidding! We actually do want to stand up a routing service like what FOSSGIS runs for OpenStreetMap: OpenHistoricalMap/issues#695. @ianthetechie and I think it should be feasible, but please chime in with any concerns or juicy scope-creep you have in mind.

**Comment by 1ec5** [comment]
The OpenStreetMap project once again applied to participate in this year’s Google Summer of Code. We’ll hear from Google this week about whether students can propose OSM-related projects. OSM usually suggests a bunch of mentor-suggested project ideas for students to choose from, but since the ideas haven’t been forthcoming this year, I’ve posted a [Valhalla–OHM team-up](https://wiki.openstreetmap.org/wiki/Google_Summer_of_Code/2024/Project_ideas#Valhalla_router) there. If OSM gets accepted and a student expresses interest in this idea, expect to hear a lot more from us.

**Reply by nilsnolde** [reply]
Sounds like a fun project:) I didn't read through the whole issue you linked in detail, but as Ian pointed out, coming up with new costings is a bit challenging and new access modes are not possible anymore soon (maybe are with lots of bit gymnastics). So that might be a bummer if you plan for that, e.g. horse (carriage). I think with a Valhalla maintainer on board it's kinda feasible, but as always, expect to do the last 20% yourself. From experience I can tell you that GSoC students have a _very hard_ time getting into routing concepts and Valhalla has a very high bar in general. For myself I chose to not participate in GSoC this year, but maybe you might have more luck with others.

**Reply by kevinkreiser** [reply]
yeah my story is i wanted to submit a project with lesser scope and less routing knowledge required but ended up running out of time.

**Reply by 1ec5** [reply]
Good point about access modes. Though for the time periods where carriage routing would be most relevant, I wonder if traffic enforcement had really been so strict that any legal access restrictions would apply. It could end up being just a costing model based on car access. 🙂 For what it’s worth, OHM maintains forks of enough OSM software that a soft fork of Valhalla wouldn’t be out of the question for, say, replacing golf cart routing with horse routing. But of course we’d want any generally applicable improvements to get contributed upstream.

**Reply by ianthetechie** [reply]
Late reply, but yeah, I think just basing things off the existing access modes would be the most logical. Unfortunately I'm in the process of using up the last bit for access modes, but for the OHM use case, you could probably just "overload" some of them internally (ex: they didn't have motorcycles or golf carts back in the day, so could reuse those and just have your API rewrite them if you care).

Maintaining your own soft fork is certainly possible; we've done that at times as well, but the costs do become significant long term, so I think going in with a "maintenance strategy" upfront to minimize this would be helpful.

I have some _super_ half-baked thoughts re: extensibility of costing models as well, but I'll write these up in a separate discussion.

**Comment by shinomiya-akirawane** [comment]
Hi, my name is Zhaoyan Dong and I am a third-year undergraduate student at UCL. I am really interested in this project and started my initial exploration of Valhalla. May I ask did you heard back from Google about whether they accept the OSM proposal? @1ec5

**Reply by 1ec5** [reply]
Hi @shinomiya-akirawane, Google did accept OSM as a mentoring organization, so you’re welcome to submit a proposal for any of [the ideas on this page](https://wiki.openstreetmap.org/wiki/Google_Summer_of_Code/2024/Project_ideas) or one of your own that’s related to either OSM or OHM. If you’d like to do something based on this OHM routing idea, please reach out to @ianthetechie.

**Reply by shinomiya-akirawane** [reply]
I will continue my exploration and reach out to Ian. Thank you so much for answering my question.

**Reply by shinomiya-akirawane** [reply]
Hi @1ec5 ,

I apologize for dropping this irrelevant comment here, but after fully exploring the Idea Scale, I feel that it's unlikely for me to complete the development of both the idea I initially mentioned and the one I want to assist with (the citation one) during the GSoC period. After careful consideration, I believe that combining the citation idea with the Intuitive EDTF date entry in OpenHistoricalMap idea would be a better choice for me. Could you please provide me with your contact information (your email address?), as you are the potential mentor for both of these ideas?

**Reply by 1ec5** [reply]
Hi @shinomiya-akirawane, thanks for your consideration. You can e-mail me at the e-mail address on the second `From:` line of [this patch](https://github.com/OpenHistoricalMap/iD/commit/aa378b0a3d922771ac4a44e7f769dc2218eeec82.patch), or you can send a private message to me on [OSM](https://www.openstreetmap.org/user/Minh%20Nguyen) or [OHM](https://www.openhistoricalmap.org/user/Minh%20Nguyen), or you can DM me at @1ec5 on [OSMUS Slack](https://slack.openstreetmap.us/). Looking forward to hearing from you.
